framework/source/class/qx/ui/core/Blocker.js

```diff
@@ -261,7 +261,9 @@ qx.Class.define("qx.ui.core.Blocker",
       this.__focusElements.push(qx.ui.core.Widget.getWidgetByElement(focusHandler.getFocus()));
 
       if (this._widget.isFocusable()) {
-        this._widget.focus();
+        setTimeout(function() {
+          this._widget.focus();
+        }.bind(this), 0);
       }
```

Comment:
The failing test awaits the focus being set to the widget right after the blocker blocks, but that is not the case anymore due to the delayed focusing.

Does it do any harm if the focusing the widget is called twice? If not we could do instead:
```javascript
      if (this._widget.isFocusable()) {
        this._widget.focus();
        qx.event.Timer.once(this._widget.focus, this._widget, 0);
      }
```
which ensures that the focus is still at the widget after the current code gives the control back to the engine.

Reply:
We had discussed that, too. Feels a bit "hackish" ;-)

Reply:
But maybe it's not more hackish than the setTimeout...

Reply:
the question is if the `this._widget` fires the focus event twice in this situation which would make a difference

Reply:
In the end it may be that we get a `focus` -> `blur` -> `focus` event sequence instead of a single `focus` event

Reply:
The interresting thing is that it doesn't differ. If you've just the `this._widget.focus()`, you'll have 6 focus changes in Chrome (due to the *mouseup*). If you add the `setTimeout` path, you've just 4 focus changes, because the *mouseup* doesn't change the initial "Button is focussed".